Fix GLTFExport Bug with Complex MorphTarget Animations.

Bug-Reaper opened the pull request:
Addresses a bug where GLTF Exporter is unable to merge/export morphtarget animations that aren't targeting the root object3d being exported.

Github Issue: https://github.com/mrdoob/three.js/issues/17579

Bug-Reaper committed:
Fix GLTFExport Bug with MorphTarget Animations which aren't aimed at the root node being exported:
Github Issue: https://github.com/mrdoob/three.js/issues/17579

donmccurdy commented on examples/js/exporters/GLTFExporter.js:
```diff
@@ -2254,7 +2254,9 @@ THREE.GLTFExporter.Utils = {
-				mergedTrack.name = '.morphTargetInfluences';
+				// We need to take into consideration that the target node for a
+				// blendshape animation might not be the root node of the animation:
+				mergedTrack.name = sourceTrack.name;
```
I think you're on the right track, but isn't `sourceTrack.name` going to be something like `widget.morphTargetInfluences[5]`? How about:

```js
mergedTrack.name = sourceTrackBinding.nodeName + '.morphTargetInfluences';
```

Bug-Reaper replied:
I thought so too. Originally I had tried doing several other assignments following the same reasonable thinking behind your suggestion. However I believe that the other properties like "nodeName" either do not have the information we need or result in bugged/failed .gltf export.

Although you are correct that `sourceTrack.name` will be equal to  `widget.morphTargetInfluences["key1"]` . I can guarantee that the code works for creating valid .gltf files with animations. (Tested with your gltf2 viewer to be error/warning free).

That  being said, I will try the exact line you suggested and if that works we can use that as it is a better answer and I do not totally understand why simply using `sourceTrack.name` works in the first place.

On an unrelated note. The current GLTFExporter will also crash if you try to merge/export animations that use integer lookup instead of their morphTarget dictionary counterparts. This should be addressed as a separate issue though.

Bug-Reaper replied:
`mergedTrack.name = sourceTrackBinding.nodeName + '.morphTargetInfluences';`

Works well. I'll update this PR accordingly.

 Interesting that using sourceTrack.name worked so well to begin with.

Bug-Reaper replied:
Change is made. Also clarified the comment to make more sense.

donmccurdy replied:
> Interesting that using sourceTrack.name worked at all to begin with...

I think the later parsing of the track name was ignoring that additional `["key"]` part, knowing it had already been merged. But it's better to not rely on that coincidence. Thanks!

Bug-Reaper replied:
Cool! Totally agreed but good to know haha!

Bug-Reaper committed:
Fix GLTFExport MorphTarget Bug

Chose a better candidate var for determining the true target node when merging morphTarget animation tracks.
Ammended comment to make more sense.

mrdoob commented:
Thanks!

Bug-Reaper commented:
Thank you too! Always a pleasure to contribute to such an amazing library!
